Added new translations, added sounds and more

KingIronMan2011 opened the pull request:
Added the translations for German, Spanish, French, Italian, Japanese, Russian and Chinese Simplified.
And also made some gitignore updates. Everything else was either reverted because I didnt want to break anything or smth else ig.

KingIronMan2011 committed:
feat: add project configuration files for Eclipse IDE

KingIronMan2011 committed:
fix: update .gitignore to include Java project files and modify MinecraftClientMixin annotation

KingIronMan2011 committed:
chore: remove Eclipse project configuration files

KingIronMan2011 committed:
fix: remove Java project files from .gitignore

KingIronMan2011 committed:
fix: add Java project files to .gitignore

KingIronMan2011 committed:
fix: add missing imports in MinecraftClientMixin for BlockColors and BlockPos

KingIronMan2011 commented:
Fixed the pulling in of spectator players

KingIronMan2011 commented:
Everything got tested and it works

KingIronMan2011 commented:
Formatted all files for code consistency

KingIronMan2011 commented:
Fixed it

KingIronMan2011 commented:
Everything is tested and working correctly.

KingIronMan2011 commented:
sound is fixed now

Mishkis commented:
yes. just by looking at it, I can already see several places where the formatting has made the code harder to read (specifically the shader .jsons). in addition, several of them seem to just be differences between the spaces used, which means that if I write new code it would be inconsistent with those choices. there are also places where the formatter added extra lines when they really just did not need to exist (putting extra lines in the mixins after @shadow and @final).

KingIronMan2011 commented:
it was just the standard formatter of VSCode and most of the times it works great and makes the code more readable. Atleast for website code.

KingIronMan2011 commented:
> no worries, just in this case the formatter did not really help. in addition, most people use intellij for java code instead of vscode, therefore using vscode conventions doesn't really make sense. realistically, the only way I will merge this is if you revert almost all of the commits and keep only the ones which add localization.

fr? :( why did I spend so much time on this then?

KingIronMan2011 commented:
yes ik but I just thought this would be more helpful

KingIronMan2011 committed:
Refactor code structure for improved readability and maintainability

KingIronMan2011 commented:
Ok so what I did is the following:
-Reverted to before I added the languages and stuff
-Added all languages back (with some extra keys currently like the sound subtitles and ItemGroups)
-Also left the sounds in there for an easier implementation later
-Fixed the issue that also people in spectator mode got sucked in by the laser

Tested built functionality and it works fine.

KingIronMan2011 committed:
docs: add README.md for Orbital Railgun mod with usage instructions and warnings

KingIronMan2011 commented:
ok

KingIronMan2011 committed:
fix: remove unused item group entries and delete obsolete sound files

KingIronMan2011 commented:
If you want you can add me on DC and I will make you one and then give you ownership to it

Mishkis commented:
> If you want you can add me on DC and I will make you one and then give you ownership to it

that will not be necessary.

KingIronMan2011 commented:
> > If you want you can add me on DC and I will make you one and then give you ownership to it
> 
> that will not be necessary.

no fr I would do it no problem. Im bored so I have enough time for it lol

Mishkis commented:
thank you for your time and effort on this pull request!

KingIronMan2011 commented:
no problem

KingIronMan2011 commented:
but would you still add me on dc? I think you write great code and also never saw a women write code lol. I think its really great to be so skilled in it. And also maybe I will make some plugins in the future and it would be nice to have someone that I could ask for some Java help. My dc username is ```kingironman2011```
